games/static/games/proto.js

```diff
@@ -199,19 +199,47 @@ class Board {
         }
     }
 
-    determineDirection() {
-        const aliensAtLeftBorder = this.squares[this.alienLocations[0]].isLeftBorder()
-        const aliensAtRightBorder = this.squares[this.alienLocations[this.alienLocations.length - 1]].isRightBorder()
-        const Left = -1
-        const Right = 1
-        if ((aliensAtLeftBorder && this.direction === Left) || (aliensAtRightBorder && this.direction === Right)) {
-            this.direction = this.width
-        } else if (this.direction === this.width) {
-            if (aliensAtLeftBorder) this.direction = Right
-            else this.direction = Left
+    determineDirection() { 
+        let direction = this.direction    
+        if ((this.aliensAtLeftBorder() && this.aliensMovingLeft()) || (this.aliensAtRightBorder() && this.aliensMovingRight())) {
+            direction = this.width
```

Comment:
what does `direction = this.width` means ?

Reply:
it means go down